Automattic/wp-calypso
# WooCommerce: Update variation component to use product redux state

This PR updates the variation types component to use the product attribute actions to update a product's attributes/variation types as the user fills out their information. Previously the component (my first) just had internal state, which has been removed in this PR & converted to a functional component.

To Test:

* `make run`
* Go to http://calypso.localhost:3000/store/products/add/
* Toggle the "has variations" toggle and make sure this works.
* Type in a variation type name like 'Color' and make sure the name field works.
* Add options like Red & Blue to make sure the options field works.
* Click "Add another variation" to make sure new rows can be added.
